Scientist4JCore/src/main/java/com/github/rawls238/scientist4j/Experiment.java

```diff
@@ -20,75 +18,63 @@
-        this(name, new HashMap<>(), raiseOnMismatch, metricRegistry);
-    }
-
-    public Experiment(String name, Map<String, Object> context, boolean raiseOnMismatch, MetricRegistry metricRegistry) {
-        this(name, context, raiseOnMismatch, metricRegistry, Objects::equals);
+    public Experiment(String name, Map<String, Object> context, boolean raiseOnMismatch, MetricsProvider<?> metricsProvider) {
+        this(name, context, raiseOnMismatch, metricsProvider, Objects::equals);
     }
 
     public Experiment(String name, Map<String, Object> context, boolean raiseOnMismatch,
-                      MetricRegistry metricRegistry, BiFunction<T, T, Boolean> comparator) {
-        this(name, context, raiseOnMismatch, metricRegistry, comparator, Executors.newFixedThreadPool(2));
+                      MetricsProvider<?> metricsProvider, BiFunction<T, T, Boolean> comparator) {
+        this(name, context, raiseOnMismatch, metricsProvider, comparator, Executors.newFixedThreadPool(2));
     }
 
     public Experiment(String name, Map<String, Object> context, boolean raiseOnMismatch,
-                      MetricRegistry metricRegistry, BiFunction<T, T, Boolean> comparator,
+                      MetricsProvider<?> metricsProvider, BiFunction<T, T, Boolean> comparator,
                       ExecutorService executorService) {
         this.name = name;
         this.context = context;
         this.raiseOnMismatch = raiseOnMismatch;
         this.comparator = comparator;
-        controlTimer = getMetrics(metricRegistry).timer(MetricRegistry.name(NAMESPACE_PREFIX, this.name, "control"));
-        candidateTimer = getMetrics(metricRegistry).timer(MetricRegistry.name(NAMESPACE_PREFIX, this.name, "candidate"));
-        mismatchCount = getMetrics(metricRegistry).counter(MetricRegistry.name(NAMESPACE_PREFIX, this.name, "mismatch"));
-        candidateExceptionCount = getMetrics(metricRegistry).counter(MetricRegistry.name(NAMESPACE_PREFIX, this.name, "candidate.exception"));
-        totalCount = getMetrics(metricRegistry).counter(MetricRegistry.name(NAMESPACE_PREFIX, this.name, "total"));
+        this.metricsProvider = metricsProvider;
+        controlTimer = getMetricsProvider().timer(NAMESPACE_PREFIX, this.name, "control");
+        candidateTimer = getMetricsProvider().timer(NAMESPACE_PREFIX, this.name, "candidate");
+        mismatchCount = getMetricsProvider().counter(NAMESPACE_PREFIX, this.name, "mismatch");
+        candidateExceptionCount = getMetricsProvider().counter(NAMESPACE_PREFIX, this.name, "candidate.exception");
+        totalCount = getMetricsProvider().counter(NAMESPACE_PREFIX, this.name, "total");
         executor = executorService;
     }
 
     /**
-     * Allow override here if extending the class, use the one passed into constructor if not null
-     * or resort to the default created one internally
+     * Allow override here if extending the class
```

Comment:
This extension point may no longer be helpful, since `MetricsProvider` is now in all the constructors.